New package: CornerPlotting v0.1.0

JuliaRegistrator opened the pull request:
- Registering package: CornerPlotting
- Repository: https://github.com/orlox/CornerPlotting.jl
- Created by: @orlox
- Version: v0.1.0
- Commit: 98ddd17bbc21879567e6f7f4f1fb9eedf7e6b9ed
- Reviewed by: @orlox
- Reference: https://github.com/orlox/CornerPlotting.jl/issues/9#issuecomment-2365145232
- Description: Simple plotting tool to produce corner plots out of samples and determine credible intervals
<!-- bf0c69308befbd3ccf2cc956ac8a4671113b158fd87a6d4e741b0173dab051a48f7dc43e593ffff4357bc647a3a835988f5e0c805f1640e4c751cc30f90d2f3c19af05f54a93fd64752cd5c692c49d2a70e560af5edfb232926d5649f9dfd96d223b3baa61b106a22b653d28a973ce1a6e0c62c646e9b9cf6217f5f72d741f1e114694e580008d6f78ccf4b005bf222c39258f5510af0ab57763fb2d350fdd9a4a0ed3db881f07fe34fc358c9c533528 -->

goerz commented:
Can you comment (also in your README or documentation) on how this compares to [PairPlots](https://github.com/sefffal/PairPlots.jl)?

@sefffal

orlox commented:
Sure thing! I needed in particular a corner plotter that could provide highest density credible intervals in both 1D and 2D. My understanding is that PairPlots uses quantiles for its credible intervals, not entirely sure what the contours represent. But also I needed a tool that could give me easy access to elements of the plot to modify, to include for instance priors. This is code I had embedded in two different projects, so it made more sense to make it more general and make it its own package.

sefffal commented:
I think you might be surprised the breadth of features and customization available in PairPlots:
* https://sefffal.github.io/PairPlots.jl/dev/api#PairPlots.Contour
* https://sefffal.github.io/PairPlots.jl/dev/guide

But I think it's okay to have duplication of packages with different APIs; there's lots of examples of this in the Julia ecosystem!

BTW for other readers, we also have the unmaintained CornerPlots.jl. It's a shame that it's squatting on such a nice name.

goerz commented:
> It's a shame that it's squatting on such a nice name.

There’s precedent for unmaintained packages being moved to a new maintainer if the original maintainer is no longer responsive. That’s one benefit of having a community-managed General registry.

By the way: all comments that do not contain [noblock] block the registration. So if you don’t want your comment to be blocking, make sure to edit it to contain [noblock]

orlox commented:
@sefffal ,

Thanks for the info, and congrats on your nicely documented package. Before doing this I did look a bit into the source code and docs of PairPlots and didn't quite find the way to do what I want. Contours are given by sigmas, so I guess one can just transform a specific fraction into its equivalent sigma. For HDIs for the 1D marginalized distribution I don't understand how that is possible. Docs mention to check the documentation for `MarginConfidenceLimits`, but that is empty, though from the source code I understand it does 16,50,84 percentiles, so 1-sigma equivalent percentiles. I also cannot find a simple way to access an individual panel (say, I want the axis plotting 'x' versus 'y' and overplot a scatter), but perhaps I missed that on the docs.

There are other differences  that are perhaps more personal preference than general use case. I want to have full control over styling (eg tickrotation is one example), and I do not want to use KDEs. But I can of course understand if people here think these differences do not merit another package in the general registry. I can instead just duplicate this code across projects then, which is not an enormous nuisance.

PS: I do find it interesting that we are both from astro, guess there is a particular love in our field for these plots :). It is also a pity that `CornerPlot` already exists. Following typical julian package naming I was thinking of `CornerPlots`, but that is too similar to that already existing package.

[noblock]

goerz commented:
>  I was thinking of `CornerPlots`, but that is too similar to that already existing package

The bot will certainly flag it, and usually, I have a pretty hard rule that existing registered packages block variations of the package name that differ only by the plural "s". In this case, since `CornerPlot` is apparently abandoned (we should check in with @kilianbreathnach), I might be open to overriding that rule. I'm not sure that I would be comfortable making that decision by myself, though, so this should be discussed in the `#pkg-registration` channel on [Slack](https://julialang.org/slack/) before renaming `CornerPlotting` to `CornerPlots`. It would be a step that's pretty close to taking `CornerPlot` away from @kilianbreathnach and transferring it to @orlox, which isn't something that should be done lightly.

I would also point out that `CornerPlots` is a very generic name, and that such a name would imply some higher expectations towards @orlox as a maintainer. The package should be sufficiently broad in scope to serve as the go-to package for people wanting to do corner plots in Julia. We would very much want to avoid a `CornerPlots` becoming unmaintained. It would help if there was some kind of backing of the package by someone with a permanent position, and/or the package was being hosted by a GitHub organization with multiple owners, as opposed to a personal account.

In comparison, `CornerPlotting` isn't *quite* as nice as a name, so I'd be putting a little less scrutiny on it. If you would rather not take on the extra responsibility and the (potentially prolonged) process of clearing the `CornerPlots` name, it might be best to stick to `CornerPlotting`.

[noblock]

orlox commented:
I'm ok with keeping the current name. I do expect to regularly maintain and update it, as it is a tool I need for other projects that are long running (meaning at least 10-year with secured funding). But I also expect to keep it's scope smaller than PairPlots, the idea in here is to provide simple plots for users to add on top.

[noblock]

JuliaRegistrator committed:
New package: CornerPlotting v0.1.0

UUID: 4a808bcd-0da0-4a1c-a768-2071018375da
Repo: https://github.com/orlox/CornerPlotting.jl.git
Tree: f2c1b0cefb3ef9d2ecf7864dc12139883137491c

Registrator tree SHA: 17aec322677d9b81cdd6b9b9236b09a3f1374c6a

orlox commented:
Just updated the package with actual documentation, as well as a mention to the alternatives.

[noblock]
